pints/_mcmc/_differentialEvolution.py

```diff
@@ -0,0 +1,578 @@
+
+        # Convert 3d array to list of lists
+        samples = [chains[:, i, :] for i in range(0, self._num_chains)]
+
+        # Return generated chain
+        return samples
+
+    def set_burn_in(self, burn_in):
+        """
+        Sets the number of iterations to discard as burn-in in the next run.
+        """
+        burn_in = int(burn_in)
+        if burn_in < 0:
+            raise ValueError('Burn-in rate cannot be negative.')
+        self._burn_in = burn_in
+
+    def set_iterations(self, iterations):
+        """
+        Sets the total number of iterations to be performed in the next run
+        (including burn-in and non-adaptive iterations).
+        """
+        iterations = int(iterations)
+        if iterations < 0:
+            raise ValueError('Number of iterations cannot be negative.')
+        self._iterations = iterations
+
+    def set_thinning_rate(self, thinning):
+        """
+        Sets the thinning rate. With a thinning rate of *n*, only every *n-th*
+        sample will be stored.
+        """
+        thinning = int(thinning)
+        if thinning < 1:
+            raise ValueError('Thinning rate must be greater than zero.')
+        self._thinning_rate = thinning
+
+    def thinning_rate(self):
+        """
+        Returns the thinning rate that will be used in the next run. A thinning
+        rate of *n* indicates that only every *n-th* sample will be stored.
+        """
+        return self._thinning_rate
+
+    def set_gamma(self, gamma):
+        """
+        Sets the gamma coefficient used in updating the position of each
+        chain.
+        """
+        if gamma < 0:
+            raise ValueError('Gamma must be non-negative.')
+        self._gamma = gamma
+
+    def set_b(self, b):
+        """
+        Sets the normal scale coefficient used in updating the position of each
+        chain.
+        """
+        if b < 0:
+            raise ValueError('normal scale coefficient must be non-negative.')
+        self._b = b
+
+    def set_num_chains(self, num_chains):
+        """
+        Sets the number of chains to evolve
+        """
+        if num_chains < 10:
+            raise ValueError('This method works best with many chains (>>10,'
+                             + 'typically).')
+        self._num_chains = num_chains
+
+
+def differential_evolution_mcmc(log_likelihood, x0, sigma0=None):
+    """
+    Runs an differential evolution MCMC routine with the default parameters.
+    """
+    return DifferentialEvolutionMCMC(log_likelihood, x0, sigma0).run()
+
+
+class DreamMCMC(pints.MCMC):
+    """
+    *Extends:* :class:`MCMC`
+
+    Uses differential evolution adaptive Metropolis (DREAM)
+    MCMC as described in [1] to do posterior sampling
+    from the posterior.
+
+    In each step of the algorithm N chains are evolved
+    using the following steps:
+
+    1. Select proposal:
+
+    x_proposed = x[i,r] + (1 + e) * gamma(delta, d, p_g) *
```

Comment:
Can we possibly use LaTex format? Probably hard to read these equations in the documentation. (see above -- line 217)